packages/app/src/App.tsx

```diff
@@ -48,392 +51,447 @@ import {
-                          setValue={(value) => {
-                            setFlameFunctions(
-                              i(),
-                              'variations',
-                              j(),
-                              produce((draft) => {
-                                if (!isParametricType(draft)) {
+                      >
+                        <For each={Object.keys(examples) as ExampleID[]}>
+                          {(exampleId) => (
+                            <option value={exampleId}>{exampleId}</option>
+                          )}
+                        </For>
+                      </select>
+                    </>
+                  ),
+                  options: {
+                    cancel: (props) => (
+                      <button onClick={props.onClick}>Cancel</button>
+                    ),
+                    load: (props) => (
+                      <button onClick={props.onClick}>Load</button>
+                    ),
+                  },
+                })
+                if (result === 'cancel') {
+                  return
+                }
+                // structuredClone required in order to not modify the original, as store in solidjs does
+                history.replace(structuredClone(examples[selectedExampleId()]))
+              }}
+            >
+              Load Example
+            </button>
+          </Card>
+          <For each={flameFunctions}>
+            {(flame, i) => (
+              <Card>
+                <button
+                  class={ui.deleteFlameButton}
+                  onClick={() => {
+                    setFlameFunctions((draft) => {
+                      draft.splice(i(), 1)
+                    })
+                  }}
+                >
+                  <Cross />
+                </button>
+                <Slider
+                  label="Probability"
+                  value={flame.probability}
+                  min={0}
+                  max={1}
+                  step={0.001}
+                  onInput={(probability) => {
+                    setFlameFunctions((draft) => {
+                      draft[i()]!.probability = probability
+                    })
+                  }}
+                  formatValue={(value) =>
+                    formatPercent(value / totalProbability())
+                  }
+                />
+                <For each={flame.variations}>
+                  {(variation, j) => (
+                    <>
+                      <select
+                        class={ui.varInputType}
+                        value={variation.type}
+                        onInput={(ev) => {
+                          const type = ev.target.value
+                          if (!isVariationType(type)) {
+                            return
+                          }
+                          setFlameFunctions((draft) => {
+                            draft[i()]!.variations[j()] = getVariationDefault(
```

Comment:
Perhaps in a refactor, but is it possible to make more sensible names for this signals i(), j(), or is this a convention in SolidJS?

Reply:
No this is just nested loops 😀

Reply:
Yeah, I was super lazy here